### PR TITLE
Migrate additional variable initilializations from Mavlink constructor list to declarations

### PR DESCRIPTION
**Pull request by mcsauder**
**Describe problem solved by the proposed pull request**
This PR migrates an additional set of var initializations from the Mavlink class constructor list to the variable declarations following the work of PR #11610.  I will finish off this migration work in this class with one more PR after this work is accepted.

**Additional context**
See also PR #11610.

Please let me know if you have any questions on this PR.

-Mark

### CODE REVIEW

**Comment by mcsauder** [comment]
@bresch, could you help me with this? :)

File: src/modules/mavlink/mavlink_main.h [hunk 1]
```diff
@@ -589,15 +589,16 @@ class Mavlink : public ModuleParams
 	 * logic will send parameters from the current index
 	 * to len - 1, the end of the param list.
 	 */
-	unsigned int		_mavlink_param_queue_index;
+	unsigned int		_mavlink_param_queue_index{0};
+
+	bool			mavlink_link_termination_allowed{false};
```
Comment (bresch) [inline src/modules/mavlink/mavlink_main.h]: that one should also start with an underscore, right?

Comment (mcsauder) [reply src/modules/mavlink/mavlink_main.h]: I'm glad you wanted that changed.  Done!  I also deleted 3 unnecessary struct specifiers in the same commit. Thank you for your review!

**Comment by mcsauder** [comment]
Thanks @bresch !

**Pull request merged**